src/System.Net.Security/src/System/Net/Security/SslConnectionInfo.Unix.cs

```diff
@@ -2,101 +2,25 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using Microsoft.Win32.SafeHandles;
+using System.Collections.Generic;
+using System.Diagnostics;
 using System.Security.Authentication;
 
 namespace System.Net.Security
 {
-    internal partial class SslConnectionInfo 
+    internal partial class SslConnectionInfo
     {
-        public SslConnectionInfo(SafeSslHandle sslContext)
+        private void MapCipherSuite(TlsCipherSuite cipherSuite)
         {
-            Protocol = (int)MapProtocolVersion(Interop.Ssl.SslGetVersion(sslContext));
+            TlsCipherSuite = cipherSuite;
 
-            int dataCipherAlg;
-            int keyExchangeAlg;
-            int dataHashAlg;
-            int dataKeySize;
-            int dataHashKeySize;
-
-            if (!Interop.Ssl.GetSslConnectionInfo(
-                sslContext,
-                out dataCipherAlg,
-                out keyExchangeAlg,
-                out dataHashAlg,
-                out dataKeySize,
-                out dataHashKeySize))
-            {
-                throw Interop.OpenSsl.CreateSslException(SR.net_ssl_get_connection_info_failed);
-            }
-
-            DataCipherAlg = dataCipherAlg;
-            KeyExchangeAlg = keyExchangeAlg;
-            DataHashAlg = dataHashAlg;
-            DataKeySize = dataKeySize;
-            DataHashKeySize = dataHashKeySize;
-
-            //Openssl does not provide a way to return an exchange key size.
-            //It internally does calculate the key size before generating key to exchange
-            //It is not a constant (Algorthim specific) either that we can hardcode and return. 
+            TlsCipherSuiteData data = TlsCipherSuiteData.GetCipherSuiteData(cipherSuite) ?? default(TlsCipherSuiteData);
```

Comment:
This is the only call site for GetCipherSuiteData, and it's turning a null into a default.  Why not just have GetCipherSuiteData return a `TlsCipherSuiteData` instead of `TlsCipherSuiteData?` and have it return `default` instead of `null`?

Reply:
I initially have done it because we wanted cipher suite policy to match Windows and in that case this was relevant - right now since we decided on allow list I'll move that logic there